dali/pipeline/workspace/workspace.h

```diff
@@ -145,12 +145,54 @@ class WorkspaceBase : public ArgumentWorkspace {
   /**
    * @brief Returns the number of inputs.
    */
-  inline int NumInput() const { return input_index_map_.size(); }
+  inline int NumInput() const {
+    return input_index_map_.size();
+  }
 
   /**
    * @brief Returns the number of outputs.
    */
-  inline int NumOutput() const { return output_index_map_.size(); }
+  inline int NumOutput() const {
+    return output_index_map_.size();
+  }
+
+  /**
+   * Returns shape of input at given index
+   * @return TensorShape<> for SampleWorkspace, TensorListShape<> for other Workspaces
+   */
+  template <typename Backend>
+  auto GetInputShape(int input_idx) const {
+    const auto& input = *InputHandle(input_idx, Backend{});
+    return input.shape();
+  }
+
+  /**
+   * Returns batch size for given input
+   */
+  int get_input_batch_size(int input_idx) const {
```

Comment:
It's a nitpick, but why are these functions in snake_case while others in this class are in PascalCase? Not that I love PascalCase or sth, but still, these functions are not that different.

Reply:
I've written in the description, `GetBatchSize` was taken and I needed unique name to grep through it just in case. I guess `GetInputBatchSize` is unique in the code base. I can change it if you want, but this I'll do after all the green ticks necessary. I really don't like to yet again rsync me whole code base and rebuild everything, it's just a waste of time ;)